src/platform/common/stm32/config_flash.c

```diff
@@ -0,0 +1,473 @@
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include "platform.h"
+#include "drivers/system.h"
+#include "config/config_streamer.h"
+
+#if defined(CONFIG_IN_FLASH)
+
+#if defined(STM32F745xx) || defined(STM32F746xx) || defined(STM32F765xx)
+/*
+Sector 0    0x08000000 - 0x08007FFF 32 Kbytes
+Sector 1    0x08008000 - 0x0800FFFF 32 Kbytes
+Sector 2    0x08010000 - 0x08017FFF 32 Kbytes
+Sector 3    0x08018000 - 0x0801FFFF 32 Kbytes
+Sector 4    0x08020000 - 0x0803FFFF 128 Kbytes
+Sector 5    0x08040000 - 0x0807FFFF 256 Kbytes
+Sector 6    0x08080000 - 0x080BFFFF 256 Kbytes
+Sector 7    0x080C0000 - 0x080FFFFF 256 Kbytes
+
+F7X5XI device with 2M flash
+Sector 8    0x08100000 - 0x0813FFFF 256 Kbytes
+Sector 9    0x08140000 - 0x0817FFFF 256 Kbytes
+Sector 10   0x08180000 - 0x081BFFFF 256 Kbytes
+Sector 11   0x081C0000 - 0x081FFFFF 256 Kbytes
+*/
+
+uint32_t getFLASHSectorForEEPROM(void)
+{
+    if ((uint32_t)&__config_start <= 0x08007FFF)
+        return FLASH_SECTOR_0;
+    if ((uint32_t)&__config_start <= 0x0800FFFF)
+        return FLASH_SECTOR_1;
+    if ((uint32_t)&__config_start <= 0x08017FFF)
+        return FLASH_SECTOR_2;
+    if ((uint32_t)&__config_start <= 0x0801FFFF)
+        return FLASH_SECTOR_3;
+    if ((uint32_t)&__config_start <= 0x0803FFFF)
+        return FLASH_SECTOR_4;
+    if ((uint32_t)&__config_start <= 0x0807FFFF)
+        return FLASH_SECTOR_5;
+    if ((uint32_t)&__config_start <= 0x080BFFFF)
+        return FLASH_SECTOR_6;
+    if ((uint32_t)&__config_start <= 0x080FFFFF)
+        return FLASH_SECTOR_7;
+#if defined(STM32F765xx)
+    if ((uint32_t)&__config_start <= 0x0813FFFF)
+        return FLASH_SECTOR_8;
+    if ((uint32_t)&__config_start <= 0x0817FFFF)
+        return FLASH_SECTOR_9;
+    if ((uint32_t)&__config_start <= 0x081BFFFF)
+        return FLASH_SECTOR_10;
+    if ((uint32_t)&__config_start <= 0x081FFFFF)
+        return FLASH_SECTOR_11;
+#endif
+
+    // Not good
+    while (1) {
+        failureMode(FAILURE_CONFIG_STORE_FAILURE);
+    }
+}
+
+#elif defined(STM32F722xx)
+/*
+Sector 0    0x08000000 - 0x08003FFF 16 Kbytes
+Sector 1    0x08004000 - 0x08007FFF 16 Kbytes
+Sector 2    0x08008000 - 0x0800BFFF 16 Kbytes
+Sector 3    0x0800C000 - 0x0800FFFF 16 Kbytes
+Sector 4    0x08010000 - 0x0801FFFF 64 Kbytes
+Sector 5    0x08020000 - 0x0803FFFF 128 Kbytes
+Sector 6    0x08040000 - 0x0805FFFF 128 Kbytes
+Sector 7    0x08060000 - 0x0807FFFF 128 Kbytes
+*/
+
+uint32_t getFLASHSectorForEEPROM(void)
+{
+    if ((uint32_t)&__config_start <= 0x08003FFF)
```

Comment:
(__config_start < 0x08000000 is not checked)

Reply:
Will clean up in a future PR